src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs

```diff
@@ -1091,6 +1125,57 @@ private void WriteContentFilesToPreprocess()
                     });
             }
 
+            private void WriteDebugSymbolsFiles()
+            {
+                WriteDebugItems(
+                    p => p.RuntimeAssemblies,
+                    MetadataKeys.PdbExtension);
+            }
+
+            private void WriteDebugXmlFiles()
+            {
+                if (_task.PublishReferencesDocumentationFiles)
+                {
+                    WriteDebugItems(
+                        p => p.CompileTimeAssemblies,
+                        MetadataKeys.XmlExtension);
+                }
+            }
+
+            private void WriteDebugItems<T>(
+                Func<LockFileTargetLibrary, IEnumerable<T>> getAssets,
+                string extension)
+                where T : LockFileItem
+            {
+                foreach (var library in _runtimeTarget.Libraries)
+                {
+                    if (!library.IsPackage())
+                    {
+                        continue;
+                    }
+
+                    foreach (T asset in getAssets(library))
+                    {
+                        if (asset.IsPlaceholderFile() || !asset.Properties.ContainsKey(MetadataKeys.RelatedProperty))
+                        {
+                            continue;
+                        }
+
+                        string itemSpec = _packageResolver.ResolvePackageAssetPath(library, asset.Path);
+
+                        asset.Properties.TryGetValue(MetadataKeys.RelatedProperty, out string relatedExtensions);
+
+                        foreach (string fileExtension in relatedExtensions.Split(RelatedPropertySeparator))
+                        {
+                            if (fileExtension.ToLower() == extension)
```

Comment:
@dsplaisted 
My understanding is that only <assemblyName>.pdb, <assemblyName>.xml

I am not sure if we should handle xml extension like the in the example here:
"related": ".Http.xml;.HttpListener.xml;.Mail.xml;.NameResolution.xml;.NetworkInformation.xml;.Ping.xml;.Primitives.xml;.Requests.xml;.Security.xml;.ServicePoint.xml;.Sockets.xml;.WebClient.xml;.WebHeaderCollection.xml;.WebProxy.xml;.WebSockets.Client.xml;.WebSockets.xml"

Reply:
This may be a bug in NuGet (or a bug in how we specc'ed the behaivor).  Where did you get this example from?